src/main/ts/utils/Git.ts

```diff
@@ -117,3 +117,21 @@ const detectGitUrlFromDir = async (dir: string): Promise<string> => {
 
 export const resolveGitUrl = async (gitUrlArg: Option<string>, workingDirArg: string): Promise<string> =>
   O.isSome(gitUrlArg) ? gitUrlArg.value : await detectGitUrlFromDir(workingDirArg);
+
+const isWorkingDirDirty = async (git: SimpleGit) => {
+  const diff = await git.diffSummary([ 'HEAD' ]);
```

Comment:
I've been bitten by the git CLI too many times, so I want to check. Will this notice:
* Changes that are already staged for the next commit
* Untracked files that aren't covered by your gitignore

Reply:
When I was reviewing this I wondered the same thing, as I wondered if we'd be better using the `status` command instead. I did verify the first one is handled in this case, however I did not check the second case.

Reply:
This will not detect untracked files, I don't think we should be concerned about that?